### PR TITLE
MULE-19301: Migrate param validations in `ParametersResolver` and `RequiredParameterValueResolverWrapper`

### PR DESCRIPTION
**Pull request by elrodro83**
* Also refactor validations tests

### CODE REVIEW

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/DynamicConfigWithStatefulOperationConfigurationOverrideTestCase.java [hunk 1]
```diff
@@ -6,23 +6,41 @@
...
   protected void additionalExceptionAssertions(ExpectedException expectedException) {
     expectedException.expectMessage("Component 'implicit:get-enriched-name' at implicitConfig/processors/0 uses a dynamic "
         + "configuration and defines configuration override parameter 'optionalWithDefault' which is "
         + "assigned on initialization. That combination is not supported. Please use a non dynamic "
         + "configuration or don't set the parameter.");
   }
 
+  @Test
+  public void fail() {
+    Assert.fail("Config should have failed to parse");
```
Comment (fsgonz) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/DynamicConfigWithStatefulOperationConfigurationOverrideTestCase.java:44]: static import.

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/connection/ConnectionProviderObjectFactory.java [hunk 2]
```diff
@@ -54,6 +56,9 @@
   private PoolingProfile poolingProfile = null;
   private ReconnectionConfig reconnectionConfig = ReconnectionConfig.getDefault();
 
+  @Inject
```
Comment (fsgonz) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/connection/ConnectionProviderObjectFactory.java:59]: where is this being used?

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/DynamicConfigWithStatefulOperationConfigurationOverrideTestCase.java [hunk 3]
```diff
@@ -6,23 +6,41 @@
...
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.test.module.extension.AbstractExtensionFunctionalTestCase;
 
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
-public class DynamicConfigWithStatefulOperationConfigurationOverrideTestCase extends InvalidExtensionConfigTestCase {
+// TODO MULE-19352 migrate this test to InvalidExtensionConfigTestCase
```
Comment (fsgonz) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/DynamicConfigWithStatefulOperationConfigurationOverrideTestCase.java:17]: use an appropriate jdoc and allure and add the TODO in there.

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/connector/PetStoreExclusiveParameterRequiredWithNullExpressionTestCase.java [hunk 4]
```diff
@@ -7,49 +7,54 @@
...
   }
 
   @Override
   protected String getConfigFile() {
-    return "petstore-exclusive-parameters-required-with-null-expression.xml";
+    return "validation/petstore-exclusive-parameters-required-with-null-expression.xml";
   }
 
   @Test
+  // This cannot be validated at the AST because the exception results of a provided expression evaluating to null
```
Comment (fsgonz) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/connector/PetStoreExclusiveParameterRequiredWithNullExpressionTestCase.java:57]: use appropriate comment format for a method or add this this comment in the method.

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/connector/PetStoreExclusiveParameterRequiredWithNullExpressionTestCase.java [hunk 5]
```diff
@@ -63,6 +68,7 @@ public void getBreederOperationSuccess() throws Exception {
...
```
Comment (fsgonz) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/connector/PetStoreExclusiveParameterRequiredWithNullExpressionTestCase.java:71]: use appropriate comment format for a method or add this this comment in the method.

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/OperationExecutionTestCase.java [hunk 6]
```diff
@@ -268,6 +268,7 @@ public void operationWithStaticInlinePojoParameter() throws Exception {
   }
 
   @Test
+  // This cannot be validated at the AST because the exception results of a provided expression evaluating to null
```
Comment (fsgonz) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/OperationExecutionTestCase.java:271]: use appropriate comment format for a method or add this this comment in the method.

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/validation/InvalidExtensionConfigTestCase.java [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+  @Test
+  public void petStoreExclusiveParameterRequired() throws Exception {
+    expectedException.expect(ConfigurationException.class);
+    expectedException
+        .expectMessage("[validation/petstore-exclusive-required-parameter.xml:9]: "
+            + "Element <Breeder> requires that one of its optional parameters must be set, but all of them are missing. One of the following must be set: [mammals, birds].");
+    loadConfiguration("validation/petstore-exclusive-required-parameter.xml");
+  }
+
+  // TODO MULE-19350 migrate and adapt this test
```
Comment (fsgonz) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/validation/InvalidExtensionConfigTestCase.java:143]: use appropriate comment format for a method or add this this comment in the method.

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/validation/InvalidExtensionConfigTestCase.java [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+  @Test
+  public void configLevelOperationNegative() throws Exception {
+    expectedException.expect(InitialisationException.class);
+    expectedException
+        .expectMessage("Root component 'appleEatsBanana' defines an usage of operation 'eatBanana' which points to configuration 'apple'. The selected config does not support that operation.");
+    loadConfiguration("validation/vegan-invalid-config-for-operations.xml");
+  }
+
+  // TODO MULE-19350 migrate and adapt this test
```
Comment (fsgonz) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/validation/InvalidExtensionConfigTestCase.java:152]: use appropriate comment format for a method or add this this comment in the method.

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/resolver/RequiredParameterValueResolverWrapperTestCase.java [hunk 9]
```diff
@@ -78,6 +81,7 @@ public void nonNullValue() throws Exception {
   }
 
   @Test
+  // This cannot be validated at the AST because the exception results of a provided expression evaluating to null
```
Comment (fsgonz) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/resolver/RequiredParameterValueResolverWrapperTestCase.java:84]: use appropriate comment format for a method or add this this comment in the method.

File: runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/RequiredParametersPresent.java [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+          } else {
+            return Optional.<ValidationResultItem>empty();
+          }
+        })
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .findFirst();
+  }
+
+  // resolve the provided value for every required param to iterate and check
```
Comment (fsgonz) [inline runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/RequiredParametersPresent.java:80]: use appropriate comment format for a method or add this this comment in the method.

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/connection/ConnectionProviderObjectFactory.java (hunk 2 above)
Comment (elrodro83) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/connection/ConnectionProviderObjectFactory.java:59]: the OCS support uses it.

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/DynamicConfigWithStatefulOperationConfigurationOverrideTestCase.java (hunk 3 above)
Comment (elrodro83) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/config/DynamicConfigWithStatefulOperationConfigurationOverrideTestCase.java:17]: this is just a TODO, not a description of the test

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/validation/InvalidExtensionConfigTestCase.java (hunk 7 above)
Comment (elrodro83) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/validation/InvalidExtensionConfigTestCase.java:143]: this is just a TODO, not a description of the test

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/validation/InvalidExtensionConfigTestCase.java (hunk 8 above)
Comment (elrodro83) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/validation/InvalidExtensionConfigTestCase.java:152]: this is just a TODO, not a description of the test

**Review by fsgonz — APPROVED** [review]

**Pull request merged**